cupy/linalg/einsum.py

```diff
@@ -319,6 +339,29 @@ def reduced_binary_einsum(arr0, sub0, arr1, sub1, sub_others):
     sub_out = sub_b + sub_l + sub_r
     assert set(sub_out) <= set_others, 'operands should be reduced: unary sum'
 
+    if _use_cutensor(arr0.dtype, sub0, arr1.dtype, sub1,
+                     batch_dims, contract_dims):
+        if len(sub_out) == len(sub_others):
+            sub_out = sub_others
```

Comment:
Could you comment the reason?  It seems a hack to make the final output C-contiguous.

Reply:
That's right!

Reply:
I just noticed that no matmul nor cutensor is necessary when either/both of inputs is/are scalar...